### PR TITLE
optimal number of objects for parallel ngmix

### LINKED ISSUE #587: [NEW FEATURE] ngmix optimal parallel split
The ngmix step (option `-j 32` with `job_sp`) is run in parallel, on subsamples of galaxies per tile.
The number of subsamples (= number of jobs) *and* the number of objects per job have to be specified
as command options to `job_sp`.

These two numbers are obviously not independent. This could be replaced by just one option, where the
number of objects is computed within the script.

For this one would need to read the number of detected galaxies from the catalogue FITS file header

### PR DESCRIPTION
**Pull request by martinkilbinger**
## Summary

Added a script to get the total number of objects in the last (tile) SExtractor run output catalogue.

Used this result in `job_sp.bash` to compute the optimal number of objects per job for shape measurement.

Closes #587 

## Reviewer Checklist

> Reviewers should tick the following boxes before approving and merging the PR.

- [x] The PR targets the `develop` branch
- [x] The PR is assigned to the developer
- [x] The PR has appropriate labels
- [x] The PR is included in appropriate projects and/or milestones
- [x] The PR includes a clear description of the proposed changes
- [x] If the PR addresses an open issue the description includes "closes #<ISSUE NUMBER>"
- [x] The code and documentation style match the current standards
- [x] Documentation has been added/updated consistently with the code
- [x] All CI tests are passing
- [x] API docs have been built and checked at least once (if relevant)
- [x] All changed files have been checked and comments provided to the developer
- [x] All of the reviewer's comments have been satisfactorily addressed by the developer

### CODE REVIEW

File: scripts/python/get_number_objects.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+from astropy.io import fits
+
+from shapepipe.pipeline.run_log import get_last_dir                             
+from shapepipe.utilities import cfis                                            
+
+
+class param:
+    """General class to store (default) variables
+
+    """
```
Comment (sfarrens) [inline scripts/python/get_number_objects.py]:
```suggestion
    """General class to store (default) variables"""
```

File: scripts/python/get_number_objects.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    Returns
+    -------
+    class param
+        parameter values
+
+    """
+    p_def = param(
+        input_path  = '.',
+        input_name_base = 'final_cat',
+        hdu_num = 1,
```
Comment (sfarrens) [inline scripts/python/get_number_objects.py]:
```suggestion
        input_path='.',
        input_name_base='final_cat',
        hdu_num=1,
```

File: scripts/python/get_number_objects.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    def var_list(self, **kwds):
+        return vars(self)
+
+
+def params_default():
+    """Set default parameter values.
+
+    Parameters
+    ----------
+    None
```
Comment (sfarrens) [inline scripts/python/get_number_objects.py]: Not needed

Comment (martinkilbinger) [reply scripts/python/get_number_objects.py]: Removed.

**Pull request merged**